ChangeLog.md

```diff
@@ -20,6 +20,10 @@ See docs/process.md for more on how version tagging works.
 
 3.1.52 (in development)
 -----------------------
+- Symbols that are explictly exported using `EXPORTED_FUNCTIONS` will now always
+  generate a linker error if they are not found.  The
+  `ERROR_ON_UNDEFINED_SYMBOLS` flag does not effect these explictly exported
+  symbols. (#21020)
```

Comment:
Might some users not want the old behavior, though? That is, they want to add a function to be exported, but are ok if it isn't present and so they set `ERROR_ON_UNDEFINED_SYMBOLS` to false to avoid that error. This PR would break them IIUC.

Reply:
Do you think that is likely?  It seems like an odd requirement that it would be simpler to just not support.

Reply:
I'll see if I can make #20970 work without this

Reply:
Note that `ERROR_ON_UNDEFINED_SYMBOLS=0` doesn't currently prevent the `undefined exported symbol` warning (which defaults to being an error).   i.e. explictly exported symbols are already treated differently, so such as user would need to pass both `-sERROR_ON_UNDEFINED_SYMBOLS=0` and `-Wno-undefined` in order to allow a function that is in `EXPORTED_FUNCTIONS` to not be defined.

I guess we should probably unify `ERROR_ON_UNDEFINED_SYMBOLS` with `-Wundefined` though...